include/deal.II/numerics/matrix_tools.h

```diff
@@ -972,7 +980,11 @@ namespace MatrixTools
   /**
    * Exception
    */
-  DeclException0 (ExcBlocksDontMatch);
+  DeclExceptionMsg (ExcBlocksDontMatch,
```

Comment:
One could move it to `exceptions.h`, but I think it's just fine here. The text accurately reflects the contexts in which this exception is currently used, and I'm ok with leaving it here.